### ISSUE TITLE
Show route for Cadl

### ISSUE THREAD

**Issue by heaths** [Azure/azure-sdk-tools#4825]
Whether a service uses Azure-Core or not for Cadl, it would be great to see what the route/endpoint will be. Even when some of us get used to doing this calculation in our heads, our customers and new reviewers onboarding will take time (or, if infrequent enough, may never truly learn it well enough).

I could think of several ways this could be accomplished, such as a system message (though, perhaps green or yellow) like this:

<img width="771" alt="image" src="https://user-images.githubusercontent.com/1532486/204607083-b3eed201-37bd-44ed-b92d-a047e634949e.png">

Or, perhaps, treat it like the documentation it is: 

<img width="429" alt="image" src="https://user-images.githubusercontent.com/1532486/204607212-5e1ec2d9-5ebc-4878-9d46-18531ffb1f25.png">

**Comment by tjprescott** [comment]
@maririos why is this assigned to me?

**Comment by maririos** [comment]
@tjprescott correct me if I'm wrong, but this looks like something that could be explored from the parser perspective

**Comment by tjprescott** [comment]
I guess I was confused because the code shown is definitely not Cadl. @heaths are you saying this should be displayed for Cadl APIViews?

**Comment by maririos** [comment]
Ohh I see. Or is this that for a generated SDK you would like to see where it came from (Cadl/OpenAPI)

**Comment by heaths** [comment]
Yes, I'm suggesting it should be shown for APIView - even if they use Azure-Core. That said, my understanding is that the parser behind APIView that each language implements might have to do it, but only that parser - I don't mean the Cadl parser/compiler itself. In a way, it already does that so maybe it's useful and can help the APIView parser figure it all out.

**Comment by heaths** [comment]
Clarifying per my offline conversation with @maririos, @tjprescott (or whoever might work on this), I meant more like a "tooltip" or some sort of hint that would show the computed route. For non-Azure-Core-based Cadls, I suppose it's unnecessary because I understand they have to include a `@route`, but for Azure-Core-based Cadls, perhaps something like this (mock-up with F12 dev tools):

Non-Azure-Core (nothing much to do since the `@route` is already there):
<img width="316" alt="image" src="https://user-images.githubusercontent.com/1532486/204907868-875649c4-1f87-4e3c-8853-d29e8e959c48.png">

Azure-Core:
<img width="219" alt="image" src="https://user-images.githubusercontent.com/1532486/204908380-862e3228-c5a3-41e7-8cea-a889439959cb.png">

The comments I added (rough example) could be treated like docs or something - perhaps shown or hidden by default - so any reviewer can easily know, no matter how experienced they are with Cadl - what the routes will look like.

**Comment by tjprescott** [comment]
Got it. It is possible to do, though it would mean that the representation in APIView isn't actually what the Cadl says (although that's technically already true).

Right now the recommendation would simply be to look at the Swagger APIView for the Cadl but without the ability to easily navigate from the Cadl to said view, that's kind of annoying.

I'd like the architects to weigh in on this and if this is something they would like.

**Comment by heaths** [comment]
> but without the ability to easily navigate from the Cadl to said view, that's kind of annoying.

Right - that's mainly the issue. Having to look at the swagger feels like hitting a thumbtack with a sledgehammer. Ideally, we never have to look at swagger again. And as we onboard new reviewers - or even team self-reviews - what the route looks like may not be obvious.

/cc @JeffreyRichter @johanste @markweitzel @mikekistler
